lesson_15/person.py

```diff
@@ -0,0 +1,25 @@
+from random import randint
+from lesson_15.bank_acc import BankAccount
+
+
+class Person:
+    def __init__(self):
+        self.bank_account = BankAccount()
+        self.car = None
+        self.house = None
+
+    def work(self):
+        earnings = randint(5, 10)
+        self.bank_account.deposit(earnings)
+
+    def buy_car(self, car):
+        if self.bank_account.withdraw(car.value):
+            self.car = car
+            return True
+        return False
+
+    def buy_house(self, house):
+        if self.car is not None and self.bank_account.withdraw(house.value):
```

Comment:
methods `buy_car` & `buy_house` looks like very similar.. better to use only one method `buy` and define type of instance our product there.. After that we can define permissions and checks for type of instance and check this instance